platform/server/http/Router/index.ts

```diff
@@ -56,60 +63,68 @@ export class KibanaResponse<T> {
 }
 
 export class KibanaRequest<
-  Params extends schema.Props = {},
-  Query extends schema.Props = {},
-  Body extends schema.Props = {}
+  Params extends schema.Props,
+  Query extends schema.Props,
+  Body extends schema.Props
 > {
   readonly headers: Headers;
 
+  /**
+   * Validates the different parts of a request based on the schemas defined for
+   * the route. Builds up the actual params, query and body object that will be
+   * received in the route handler.
+   */
   static validate<
-    Params extends schema.Props = {},
-    Query extends schema.Props = {},
-    Body extends schema.Props = {}
+    Params extends schema.Props,
+    Query extends schema.Props,
+    Body extends schema.Props
   >(
-    route: Route<
+    route: RouteConfig<
       schema.ObjectSetting<Params>,
       schema.ObjectSetting<Query>,
       schema.ObjectSetting<Body>
     >,
     req: express.Request
   ): { params: Params; query: Query; body: Body } {
-    let params: Params;
-    let query: Query;
-    let body: Body;
+    const noParams = {} as Params;
+    const noQuery = {} as Query;
+    const noBody = {} as Body;
 
+    // When `validate` isn't specific on a route we return all fields as empty
     if (route.validate === undefined) {
-      params = req.params;
-      query = req.query;
-      body = req.body;
-      return { params, query, body };
+      // TODO During development, should we log this?
+
+      return {
+        params: noParams,
+        query: noQuery,
+        body: noBody
+      };
```

Comment:
Yeah, I like this better. If they don't validate, we shouldn't just return back what was passed in via the request, because that would imply that we validated that. We should return empty objects.